### PR TITLE
add insert_after / insert_before to MultiProgress

### PR DESCRIPTION
**Pull request by chris-laplante**
Also introduce internal InsertLocation enum to clarify MultiProgress:push

### CODE REVIEW

**Comment by chris-laplante** [comment]
Here's a rough cut of what I propose :)

File: src/progress_bar.rs [hunk 1]
```diff
@@ -633,10 +641,28 @@ impl MultiProgress {
...
+    /// Inserts a progress bar after an existing one.
+    ///
+    /// The progress bar added will have the draw target changed to a
+    /// remote draw target that is intercepted by the multi progress
+    /// object overriding custom `ProgressDrawTarget` settings.
+    pub fn insert_after(&self, after: &ProgressBar, pb: ProgressBar) -> ProgressBar {
+        self.push(InsertLocation::After(after), pb)
+    }
+
+    fn push(&self, pos: InsertLocation, pb: ProgressBar) -> ProgressBar {
```
Comment (djc) [inline src/progress_bar.rs:665]: Let's not call the `InsertLocation` `pos` here? That seems confusing.

File: src/progress_bar.rs [hunk 2]
```diff
@@ -558,6 +558,14 @@ impl Default for MultiProgress {
     }
 }
 
+enum InsertLocation<'a> {
```
Comment (djc) [inline src/progress_bar.rs:561]: Style nit: please move this definition down below the `impl WeakProgressBar`.

File: examples/multi-insert-relative.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
+use indicatif::{MultiProgress, ProgressBar, ProgressFinish, ProgressStyle};
```
Comment (djc) [inline examples/multi-insert-relative.rs:1]: One of my pet peeves is that indicatif already has too many examples. Do you think you could modify one of the existing examples to use `insert_before()` and/or `insert_after()`?

**Review by djc — COMMENTED** [review]
Thanks, this looks good! Some comments below.

File: src/progress_bar.rs (hunk 1 above)
Comment (chris-laplante) [reply src/progress_bar.rs]: Done

File: src/progress_bar.rs (hunk 2 above)
Comment (chris-laplante) [reply src/progress_bar.rs]: Done

File: examples/multi-insert-relative.rs (hunk 3 above)
Comment (chris-laplante) [reply examples/multi-insert-relative.rs:1]: Sure, I can modify the "multi.rs" example and will do so now. As I alluded in the previous version of this pull request, the other multi-* examples are too complicated to change to use `insert_before`/`insert_after`.

Comment (chris-laplante) [reply examples/multi-insert-relative.rs]: Done

**Comment by djc** [comment]
Thanks!

**Comment by chris-laplante** [comment]
Thank you too :)

**Pull request merged**